corehq/blobs/pillow.py

```diff
@@ -19,46 +14,28 @@
 
 class BlobDeletionProcessor(PillowProcessor):
 
-    def __init__(self, blob_db, bucket_base):
+    def __init__(self, blob_db):
         super(BlobDeletionProcessor, self).__init__()
         self.blob_db = blob_db
-        self.bucket_base = bucket_base
 
     def process_change(self, pillow_instance, change):
         if change.deleted:
-            bucket = join(self.bucket_base, change.id)
+            bucket = "{}/{}".format(change.meta.data_source_name, change.id)
```

Comment:
FYI, I looked into this a little bit today and I think it's more complicated than this. I don't think we have any clear way to deal with hard deletions in SQL, and even in couch it doesn't look like it obviously works (I don't think that hard deletions have enough information to know the doc type which means we don't know how to publish it to kafka). For the blobdb this would be fine because you only use the ID, but for other pillows it makes things complicated. I think we might need to add a "deleted" topic to kafka and then publish all hard deletions to that.

Anyway, bottom line is that I think there's a bit of design work required to sort this out and not sure how much of a priority it is so guessing this might linger in limbo for a while.